File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/render/OpenGLContextProvider.java
package org.maplibre.nativeffi.render;

import java.util.EnumSet;

/** OpenGL context provider support flag reported by the native library build. */
public enum OpenGLContextProvider {
  WGL(1),
  EGL(1 << 1);

  private final int nativeMask;

  OpenGLContextProvider(int nativeMask) {
    this.nativeMask = nativeMask;
  }

  public int nativeMask() {
    return nativeMask;
  }

  public static EnumSet<OpenGLContextProvider> fromMask(int mask) {
    var providers = EnumSet.noneOf(OpenGLContextProvider.class);
    for (var provider : values()) {
      if ((mask & provider.nativeMask) != 0) {
        providers.add(provider);
      }
    }
    return providers;
  }
}

File: bindings/kotlin/src/commonMain/kotlin/org/maplibre/nativeffi/render/OpenGLContextProvider.kt
package org.maplibre.nativeffi.render

/** OpenGL context provider support flag reported by the native library build. */
public enum class OpenGLContextProvider(public val nativeMask: Int) {
  WGL(1),
  EGL(1 shl 1);

  internal companion object {
    internal fun fromMask(mask: UInt): Set<OpenGLContextProvider> = fromMask(mask.toInt())

    internal fun fromMask(mask: Int): Set<OpenGLContextProvider> =
      entries.filterTo(mutableSetOf()) { (mask and it.nativeMask) != 0 }
  }
}
